### ISSUE TITLE
Remove or opt out curl request to obtain ip address from the hard coded website

### ISSUE THREAD

**Issue by hozmaster** [opnsense/core#5434]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x ] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [ x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**
In file /opnsense/core/src/etc/util.inc has function called get_dyns_ip() whihc make curl request to website if no otherwise IP address cannot be determine.

Function caller should be able not to do this operation if he/she determine to do so.

And change it also to support SSL if applicable

**To Reproduce**
Part of code from opnsense/core : 

```
function get_dyndns_ip($int, $ipver = 4)
{
    $ip_address = $ipver == 6 ? get_interface_ipv6($int) : get_interface_ip($int);
    if (empty($ip_address)) {
        log_error("Aborted IPv{$ipver} detection: no address for {$int}");
        return 'down';
    }
    
    // This should be opt out by function caller
    if ($ipver != 6 && is_private_ip($ip_address)) {
        /* Chinese alternative is http://ip.3322.net/ */
        $hosttocheck = 'http://checkip.dyndns.org';
        $ip_ch = curl_init($hosttocheck);
        curl_setopt($ip_ch, CURLOPT_RETURNTRANSFER, 1);
        curl_setopt($ip_ch, CURLOPT_INTERFACE, $ip_address);
        curl_setopt($ip_ch, CURLOPT_CONNECTTIMEOUT, 5);
        curl_setopt($ip_ch, CURLOPT_TIMEOUT, 30);
        curl_setopt($ip_ch, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
        $ip_result = curl_exec($ip_ch);
```

Coder(s) can opt out host check operated by this function call

**Expected behavior**

Add boolean variable to opt out host check.

**Describe alternatives you considered**

Remove host checkk from code base.

**Screenshots**

N/A

**Relevant log files**

N/A

**Additional context**

N/A

**Environment**

N/A

**Comment by AdSchellevis** [comment]
I don't think there's any caller in core, eventually we should move this to plugins as `dns/rfc2136` and `dns/dyndns` are consumers of the function and are both more or less unsupported.

**Comment by hozmaster** [comment]
Checking ip address of local interface should part of core.

**Comment by fichtner** [comment]
There’s not much incentive moving it to plugins since this function required maintenance in the past and will according to this ticket require further maintenance. Moving it out of core with two copies each future quality will decline drastically.

**Comment by fichtner** [comment]
@hozmaster maybe you want to extend your reasoning with an actual use case, thanks

**Comment by hozmaster** [comment]
I'm currently working on plugin which is going to update dynamin ip address  into GoDaddy service. King of rewrite dyndns-plugin but only support GoDaddy service. At first at least.

**Comment by AdSchellevis** [comment]
@fichtner the only reason `get_dyndns_ip()` is still in core is due to entanglement of these two plugins, which both have maintenance issues.  I rather don't want to see other plugins depending on this  function in the near future, as one day it will move out of core. (when there's an alternative mvc plugin available for dyndns, in which case there's no reason for us to stretch this any longer)

The location to check usually depends on the provider (https://github.com/ddclient/ddclient/blob/9631575ab4b96eeba26ea7ad953d9cb68c4d5d95/ddclient.in#L106-L122), which is also a reason why this doesn't belong in core by the way.

@hozmaster when developing a dyndns plugin, please make sure to avoid entanglement of services other then the ones specified in our docs https://docs.opnsense.org/development/backend.html

**Comment by fichtner** [comment]
@AdSchellevis I don't disagree with this at all. I just want to say a potential new plugin shouldn't try to disrupt the current problematic with the function and its consumers. it can copy and modify the function as required :)

**Comment by AdSchellevis** [comment]
@fichtner I know :) it can stay where it is now knowing both legacy plugins are the only reason it's there.

**Comment by NOYB** [comment]
@Hozmaster
What is GoDaddy API?  Guessing they don't support RFC 2136 updates?

As for turning off the host IP address checking.  Check this out.  It doesn't have the turn off feature specifically but it can be configured to use a check ip service of the user's choice.
It's not exact fit for what you're doing but may inspire some ideas.

DynDNS: Check IP Services - Plugin #1731
DynDNS: Check IP Services - plugin support #3971
Has merge conflicts since closed about 20.7.b.

Though my local copy is up to date, in use and working well.
https://github.com/NOYB/plugins/tree/Check_IP_Services_-_Plugin
https://github.com/NOYB/core/tree/Check_IP_Services_-_plugin_support
